src/server/game/Combat/CombatManager.cpp

```diff
@@ -48,7 +66,12 @@
         if (!a->IsHostileTo(b) && !b->IsHostileTo(a))
             return false;
     }
-    Player *playerA = a->GetCharmerOrOwnerPlayerOrPlayerItself(), *playerB = b->GetCharmerOrOwnerPlayerOrPlayerItself();
+    Creature const* creatureA = a->ToCreature();
+    Creature const* creatureB = b->ToCreature();
+    if ((creatureA && creatureA->IsTrigger()) || (creatureB && creatureB->IsTrigger()))
```

Comment:
some trigger flagged creatures can enter combat, like the controllers in VDW, BPC or Halion